Manager database Modifications

Neerpatel125 opened the pull request:
Made changes to the database schema and connected it to my backend.

Rohit-beep-droid committed:
Database: First commit

Fixed some ugly code from last time and did a small overhaul of my page to start working on database aspect of it

Commit:
Passed db object into manager router.

Commit:
Added a DJ to the timeslots schema and model. A timeslot must have an assigned DJ for the manager role.

Commit:
Added a new schema "Notes" to display producer notes for a day. Added the collection to our database as well.

Rohit-beep-droid reviewed (approved):
Looks good. We should probably get this merged.

Rohit-beep-droid committed:
Everything is hooked to the database

All project relevant data, except playlist, is being pulled from database now, no search mechanism implemented yet

Commit:
Created comments and template for where/how to use the database in my code.

Rohit-beep-droid committed:
Changed playlists schema

Moved "producer_created" attribute to its appropriate place

Commit:
Merge branch 'producer-database' of github.com:SWE-432-Group-1/Radio-Station into manager-database

Rohit-beep-droid reviewed (approved):
Looks good.

Commit:
Removed timeslot from the playlist schema.
Modified date object handling in manager files.

Commit:
Modified timeslot schema to use strings instead of date (easier for me to handle).

Commit:
Made slight changes to Notes schema. Connected code to the database.

Commit:
Added timeslot property back to Playlist.

Commit:
Made "timeslot" property of Playlists optional.

Commit:
Changes to Playlist Schema:
	- Moved the "producer_created" property into the songs list
	- timeslot property is now required

The manager creates a timeslot and assigns a DJ and Playlist to it (playlist is initialized with 0 songs in it). Deleting a timeslot removes the playlist (can be brought back using the undo button while still on the date).

Commit:
Undoing now preserves the songs list in the playlist.

Commit:
Tested and finished database connection for all functions.

Commit:
Added "dj_played" to songs list in Playlists collection. Used to show which songs the DJ played on the manager page.

Commit:
Added sessions to "index.js"

Commit:
Implemented sessions.

Commit:
The report table now shows which playlist is being displayed.

Commit:
Fixed UNDO stack being undefined in session variable.

Commit:
Added a list to show Dj names in the textbox.

Commit:
Added a button to exit the session.

Commit:
Not clearing the Undo stack until the user decides to end their session.

Commit:
Added links to the other pages in my nav bar.

Commit:
Added a very simple home page. Slightly modified manager css.

Commit:
Some final CSS changes on my page.

Commit:
Just one line to hide overlow on the home page.

Rohit-beep-droid committed:
Made dropdown and add functionality

Added the dropdown and add functionality for playlists

Rohit-beep-droid committed:
Added delete, fixed dropdown, and fixed add

Added delete functionality, and add, delete, dropdown *should* work as intended

Rohit-beep-droid committed:
Added search and a reset mechanism

Search bar works and reset repopulates the field again with list of all songs

Rohit-beep-droid committed:
Added session and a way get to home page

Can redirect to home page, maintains independent sessions, and sessions can be closed as well

Commit:
Fixed a bug with updating Overlap session variable.

Rohit-beep-droid committed:
Final commit: added links to all three pages

Added links to all three roles pages, and fixed add song and remove song song issue

Commit:
Merge branch 'producer-database' of github.com:SWE-432-Group-1/Radio-Station into manager-database
